rllib/offline/utils.py

```diff
@@ -0,0 +1,60 @@
+import tree
+
+from typing import Any, Dict
+
+
+def flatten_dict(nested: Dict[str, Any], sep="/", env_steps=0) -> Dict[str, Any]:
+    """
+    Flattens a nested dict into a flat dict with joined keys.
+
+    Note, this is used for better serialization of nested dictionaries
+    in `OfflinePreLearner.__call__` when called inside
+    `ray.data.Dataset.map_batches`.
+
+    Note, this is used to return a `Dict[str, numpy.ndarray] from the
+    `__call__` method which is expected by Ray Data.
+
+    Args:
+        nested: A nested dictionary.
+        sep: Separator to use when joining keys.
+
+    Returns:
+        A flat dictionary where each key is a path of keys in the nested dict.
+    """
+    flat = {}
+    # `dm_tree.flatten_with_path`` returns a list of `(path, leaf)` tuples.
+    for path, leaf in tree.flatten_with_path(nested):
+        # Create a single string key from the path.
+        key = sep.join(str(p) for p in path)
```

Comment:
nit: maybe nicer: `sep.join(map(str, path)))`

Reply:
Will change it!